Setup Graft Supernode (Building from Source).md

```diff
@@ -0,0 +1,135 @@
+# Setup Graft Supernode (Building from Source)
+
+## Getting Started
+These instructions will help you build and setup a Graft Supernode from Source.
+
+## Requirements
+* OS:  Ubuntu 18.04 (other linux distributions and versions may require additional steps)
+* RAM:  2GB per CPU Core
+* If you are running from a local network you will also need to forward ports 28680 & 28681 to your supernode's local IP address.
+
+## Prepare the server
+If you have not done so add a user and give it sudo rights.  We'll use graft-user, but you can use any username you prefer.
+
+````bash
+adduser graft-user
+usermod -aG sudo graft-user
+````
+
+Logout of root and log back in with your new username and password.  Once you are logged in under the new account update and upgrade the server.  If you receive a prompt "What do you want to do about modified configuration file sshd_config?" choose the first option "install the package maintainer's version"
+
+````bash
+sudo apt-get update
+sudo apt-get upgrade -y
+````
+
+## Create a swap file
+
+````bash
+free -h
+df -h
+sudo fallocate -l 4G /swapfile          # Change 4G to the proper size swap file for your server
+sudo chmod 600 /swapfile
+sudo mkswap /swapfile
+sudo swapon /swapfile
+sudo su -
+cat <<EOF >> /etc/fstab
+/swapfile none swap sw 0 0
+EOF
+exit
+````
+
+## Add Firewall
+
+
+````bash
+sudo ufw default allow outgoing
+sudo ufw default deny incoming
+sudo ufw allow ssh/tcp
+sudo ufw limit ssh/tcp
+sudo ufw allow 28600/tcp            # Load balancer http port
+sudo ufw allow 28643/tcp            # Load balancer https port
+sudo ufw allow 28680/tcp            # p2p node port
+sudo ufw allow 28680/udp            # p2p node port
```

Comment:
This one isn't needed (graftnoded only listens on tcp)

Reply:
Updated.